lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -813,7 +800,7 @@ private function doScheduleInsert($document, &$visited, $overrideIdGenerator = n
                 // TODO: Change Tracking Deferred Explicit
                 break;
             case self::STATE_REMOVED:
-                unset($this->scheduledRemovals[$oid]);
+                $this->treeOpQueue->unqueue(TreeOperation::OP_REMOVE, $oid);
```

Comment:
Not 100% sure about the validity of this. If the Document was previously removed, and then re-inserted we remove all REMOVE operations from the stack. Not sure if that could conflict somehow with MOVE or REMOVE.